src/core/typing.ts

```diff
@@ -44,23 +42,40 @@ export interface Typing extends EmitsDiscontinuities {
 
   /**
    * Get the current typers, a set of clientIds.
-   * @returns A Promise of a set of clientIds that are currently typing.
+   * @returns The set of clientIds that are currently typing.
    */
-  get(): Promise<Set<string>>;
+  get(): Set<string>;
 
   /**
-   * Start indicates that the current user is typing. This will emit a typingStarted event to inform listening clients and begin a timer,
-   * once the timer expires, a typingStopped event will be emitted. The timeout is configurable through the typingTimeoutMs parameter.
-   * If the current user is already typing, it will reset the timer and begin counting down again without emitting a new event.
+   * This will send a `typing.started` event to the server.
+   * Events are throttled according to the `heartbeatThrottleMs` room option.
+   * If an event has been sent within the interval, this operation is no-op.
+   *
+   *
+   * Calls to `keystroke()` and `stop()` are serialized and will always resolve in the correct order.
+   * - For example, if multiple `keystroke()` calls are made in quick succession before the first `keystroke()` call has
+   *   sent a `typing.started` event to the server, followed by one `stop()` call, the `stop()` call will execute
+   *   as soon as the first `keystroke()` call completes.
+   *   All intermediate `keystroke()` calls will be treated as no-ops.
+   * - The most recent operation (`keystroke()` or `stop()`) will always determine the final state, ensuring operations
+   *   resolve to a consistent and correct state.
    *
    * @returns A promise which resolves upon success of the operation and rejects with an ErrorInfo object upon its failure.
    */
 
-  start(): Promise<void>;
+  keystroke(): Promise<void>;
 
   /**
-   * Stop indicates that the current user has stopped typing. This will emit a typingStopped event to inform listening clients,
-   * and immediately clear the typing timeout timer.
+   * This will send a `typing.stopped` event to the server.
+   * If the user was not currently typing, this operation is no-op.
+   *
+   * Calls to `keystroke()` and `stop()` are serialized and will always resolve in the correct order.
+   * - For example, if multiple `keystroke()` calls are made in quick succession before the first `keystroke()` call has
+   *   sent a `typing.started` event to the server, followed by one `stop()` call, the `stop()` call will execute
+   *   as soon as the first `keystroke()` call completes.
+   *   All intermediate `keystroke()` calls will be treated as no-ops.
+   * - The most recent operation (`keystroke()` or `stop()`) will always determine the final state, ensuring operations
+   *   resolve to a consistent and correct state.
```

Comment:
This section should include multiple`@throws` tags to describe the error conditions